### PR TITLE
CloudCustodian policy now filters by regex

### PR DESCRIPTION
**Pull request by Richard-Hansen**
The CloudCustodian policy should mark resources as `NONCOMPLIANT` if the following statement evaluates to `true`:

`(Owner tag is absent) OR (Owner tag doesn't look like an email AND Owner tag doesn't contain the string 'shared')`

The regex for not being an email look-alike:
`^((?!(@.*\\.)).)*$`

The regex for not containing the string 'shared' (case insensitive):
`(?i)^((?!shared).)*$`

Any S3 bucket that matches the filter will be at the mercy of any actions associated with the CloudCustodian policy (most likely resource deletion). Just to be safe, a second set of eyes on the regex would be much appreciated to make sure there are no edge cases.

### CODE REVIEW

**Comment by xbrianh** [comment]
@Richard-Hansen Would it be possible to add tests for these regex patterns? Examples of positive and negative matches for each regex would be useful.

**Comment by Richard-Hansen** [comment]
> @Richard-Hansen Would it be possible to add tests for these regex patterns? Examples of positive and negative matches for each regex would be useful.

Added regex tests.

File: utils/helpers.py [hunk 1]
```diff
@@ -18,3 +18,11 @@ def create_config_policy_resource_name(policy_prefix: str, resource_type:str) ->
...
 def create_deployed_config_policy_resource_name(policy_prefix: str, resource_type:str) -> str:
     return "custodian-" + create_config_policy_resource_name(policy_prefix, resource_type)
+
+
+def owner_tag_email_regex() -> str:
+    return "^((?!(@.*\\.)).)*$"
+
+
+def owner_tag_shared_regex() -> str:
+    return "(?i)^((?!shared).)*$"
```
Comment (xbrianh) [inline utils/helpers.py:28]: nit: consider defining these with constants instead of functions

**Review by xbrianh — APPROVED** [review]
lgtm

**Review by DailyDreaming — APPROVED** [review]
LGTM as well.

File: utils/helpers.py (hunk 1 above)
Comment (Richard-Hansen) [reply utils/helpers.py:28]: There's some overall refactoring that might be in order at this point. I'm going to move forward for now, but create a ticket to do some OOP changes down the line.
